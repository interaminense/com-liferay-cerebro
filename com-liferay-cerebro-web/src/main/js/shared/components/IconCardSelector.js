import autobind from 'autobind-decorator';
import Component, {Config} from 'metal-jsx';

import Icon from 'shared/components/Icon';

class IconCard extends Component {
	@autobind
	handleClick() {
		const {onSelect, value} = this.props;

		onSelect(value);
	}

	render() {
		const {selected, symbol} = this.props;

		return (
			<li class={selected ? 'selected' : ''} onClick={this.handleClick}>
				<Icon size="lg" symbol={symbol} />
			</li>
		);
	}
}

IconCard.PROPS = {
	onSelect: Config.func(),
	selected: Config.bool(),
	symbol: Config.string(),
	value: Config.number()
};

class IconCardSelector extends Component {
	render() {
		const {items, onChange, selected} = this.props;

		return (
			<ul class="icon-card-selector-root">
				{items.map(({symbol, value}) => (
					<IconCard
						key={value}
						onSelect={onChange}
						selected={selected === value}
						symbol={symbol}
						value={value}
					/>
				))}
			</ul>
		);
	}
}

IconCardSelector.PROPS = {
	items: Config.array(),
	onChange: Config.func(),
	selected: Config.number()
};

export default IconCardSelector;