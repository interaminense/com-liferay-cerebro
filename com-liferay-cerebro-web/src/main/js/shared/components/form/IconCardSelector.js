import Component, {Config} from './Component';
import {uniqueId} from 'lodash';

import IconCardSelector from 'shared/components/IconCardSelector';

class FormIconCardSelector extends Component {
	created() {
		this._id = uniqueId('FormIconCardSelector');

		const {selected} = this.props;

		if (selected) {
			this.state.value = selected;
		}
	}

	syncSelected(newVal) {
		this.state.value = newVal;
	}

	render() {
		const {selected} = this.props;

		return <IconCardSelector {...this.otherProps()} selected={selected} />;
	}
}

FormIconCardSelector.PROPS = {
	selected: Config.number()
};

FormIconCardSelector.STATE = {
	value: Config.number()
};

export default FormIconCardSelector;