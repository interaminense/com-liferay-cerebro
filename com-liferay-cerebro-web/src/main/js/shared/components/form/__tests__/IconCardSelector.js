import IconCardSelector from '../IconCardSelector';

describe('IconCardSelector', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new IconCardSelector({
			items: [{symbol: 'faro-contacts-card-layout-image-left', value: 0}],
			name: 'foo',
			selected: 0
		});

		expect(component).toMatchSnapshot();
	});
});