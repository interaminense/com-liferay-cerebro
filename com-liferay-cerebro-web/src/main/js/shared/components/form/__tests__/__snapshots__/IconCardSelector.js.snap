// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconCardSelector should render 1`] = `
<ul class="icon-card-selector-root">
  <li class="selected">
    <svg class="icon-root lexicon-icon lexicon-icon-faro-contacts-card-layout-image-left icon-size-lg">
      <use xlink:href="/images/lexicon/icons.svg#faro-contacts-card-layout-image-left"></use>
    </svg>
  </li>
</ul>
`;
